### PR TITLE
Update gh-pages with a circleci job

### PR DESCRIPTION
**Pull request by jdahm**
## Purpose

Updates the documentation on the `gh-pages` for every merge to `main`.

TODO:
- [x] Check that this works
- [x] Add credentials
- [x] Ensure it only executes on commits on `main`

## Infrastructure changes:

- Adds a new CircleCI job.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -469,6 +469,39 @@ jobs:
...
+            pip install -r requirements_docs.txt -c constraints.txt
+      - run:
+          name: Make docs
+          command: |
+            . venv/bin/activate
+            make docs
+      - run:
+          name: Move into place
+          command: |
+            git clone --branch gh-pages git@github.com:ai2cm/pace.git gh-pages
```
Comment (mcgibbon) [inline .circleci/config.yml:494]: Can you merge this plan with the doctest plan? These should only be pushed if doctests pass anyways.

Comment (jdahm) [reply .circleci/config.yml:494]: As far as I can tell these two plans are doing different things, and should be run at different times, so I'm not sure what commonality there is to merge them. The new plan `push_docs` builds and pushes docs when a PR is merged to `main`, while the `doctest` plan tests that the sphinx docs build and the snippets run without error.

Comment (jdahm) [reply .circleci/config.yml:494]: Right, this will only run if the `doctest` plan passes, since that is a branch protection requirement.

**Review by elynnwu — APPROVED** [review]
lgtm

**Pull request merged**